### ISSUE TITLE
log-loss vs cross-entropy

### ISSUE THREAD

**Issue by atiorh** [google-deepmind/language_modeling_is_compression#5]
Hi @anianruoss, great work and thanks for sharing the code to reproduce the experiments!

I am having trouble understanding why the Transformer models are not trained with standard softmax cross entropy loss (with cross-vocabulary normalization) but rather trained to [maximize the marginal unnormalized logit of the observed/correct tokens](https://github.com/google-deepmind/language_modeling_is_compression/blob/2d0e51753684d12af8d99ff532d2cfccc61be0ef/train.py#L46) because this objective doesn’t enforce a high (cross-vocabulary normalized) likelihood prediction for the correct tokens as the average unnormalized logit value drifts higher with this loss. The paper also talks about the loss as: “However, Eq. (2) is exactly the same objective used to train current foundation models, i.e., the log-loss” (Section 2) but current foundation models (autoregressive LMs) indeed use standard softmax cross entropy unless I am grossly mistaken ([example](https://github.com/karpathy/nanoGPT/blob/master/model.py#L187)).

On the other hand, during your actual compression code, I do see that you are [taking the exp](https://github.com/google-deepmind/language_modeling_is_compression/blob/2d0e51753684d12af8d99ff532d2cfccc61be0ef/compressors/language_model.py#L98) (which interprets model outputs as implicit log) and [normalizing the outputs](https://github.com/google-deepmind/language_modeling_is_compression/blob/2d0e51753684d12af8d99ff532d2cfccc61be0ef/compressors/language_model.py#L107) of the trained Transformer model across the vocabulary into the corresponding CDF.

Could you please help me resolve this conflict?

**Comment by anianruoss** [comment]
Yes, thank you for raising this issue. We use the log-loss in the paper but forgot to add it to the open-sourced code. We fixed this now.
